servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ClientStrategyInfluencerChainBuilder.java

```diff
@@ -50,21 +54,24 @@ void add(HttpLoadBalancerFactory<?> lb) {
     }
 
     void add(ConnectionFactoryFilter<?, FilterableStreamingHttpConnection> connectionFactoryFilter) {
-        connFactoryChain =
-                connFactoryChain.merge(HttpExecutionStrategy.from(connectionFactoryFilter.requiredOffloads()));
+        connFactoryChain = connFactoryChain.merge(connectionFactoryFilter.requiredOffloads());
     }
 
     void add(StreamingHttpConnectionFilterFactory connectionFilter) {
-        connFilterChain =
-                connFilterChain.merge(HttpExecutionStrategy.from(connectionFilter.requiredOffloads()));
+        connFilterChain = connFilterChain.merge(connectionFilter.requiredOffloads());
     }
 
     HttpExecutionStrategy buildForClient(HttpExecutionStrategy transportStrategy) {
-        return clientChain.merge(buildForConnectionFactory(transportStrategy));
+        return transportStrategy.merge(clientChain.merge(connFilterChain.merge(
+                HttpExecutionStrategy.from(buildForConnectionFactory()))));
```

Comment:
#### How offloading is applied/combined
Currently the strategy for the connection factory and transport are being combined together. This means that if the connection factory blocks, then each request will be offloaded. However the connection factory is unlikely to be invoked on each request bcz the LB is assumed to pool connections.  Is it the intention to always combine the strategies into a HttpExecutionStrategy, and apply it at the transport level? If so, this `buildForClient` method is only used when converting to other API types in `FilterableClientToClient` so I'm not sure the offloading will be used in the expected spot.

Can you describe how all offloading strategies are expected to combine/interact on the client side, and how API optimizations are applied if default offloading is used? (`ClientFilter -> LoadBalancer -> ConnectionFactoryFilter -> ConnectionFilter -> Transport`)

#### Fallback method for combining
`HttpExecutionStrategy.from` fallback if a `HttpExecutionStrategy` is not returned will offload more aggressively than necessary (assuming we need to combine request-path offloading with connection creation offloading). For example for `ConnectFactory` only needs to offload when creating a connection. I wonder when we expect this condition to happen, and if we can rely upon the `merge` method of the runtime types to return the correct merge (or fallback to offload all if not implemented correctly, this is the solution we previous discussed).

Reply:
Briefly, the current combination of strategies offered is because connection filters can alter the http strategy required. The is independent of their needs for offloading of the connection establishment step.